Documentation/Diagnostics/PH2116.md

```diff
@@ -0,0 +1,58 @@
+# PH2116: Avoid ArrayList
+
+|  |  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2116 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidArrayListAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/master/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidArrayListAnalyzer.cs)
+| CodeFix  | Available |
+| Severity | Error |
+
+## Introduction
+
+The `ArrayList` should be avoided. It doesn't use generics and therefore suffers from performance issues when used with structs. Structs need to be [boxed](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/types/boxing-and-unboxing) into a class on every call to `ArrayList`, which impacts the performance negatively.
+
+Microsoft discourages the use of `ArrayList` for this reason, as can be read on [MS Learn](https://learn.microsoft.com/en-us/dotnet/api/system.collections.arraylist?view=net-7.0#remarks).
```

Comment:
I like the external links

Reply:
We should make an effort to find these links, so the user can read in more detail there